pages/recipes/api-webauthn/profile.tsx

```diff
@@ -0,0 +1,154 @@
+                  <>
+                    <Image alt="Lock" src={lock} width={100} />
+                    <p>
+                      Super secret profile information is secured by two factor authentication. To unlock this area
+                      complete the WebAuthn flow.
+                    </p>
+                    {hasRegisteredWebAuthnDevice ? (
+                      <>
+                        <button onClick={handleAuthenticate}>Authenticate</button>
+                      </>
+                    ) : (
+                      <>
+                        <p>You have not yet registered a device for Webauthn as a second factor.</p>
+                        <button onClick={handleRegister}>Register now</button>
+                      </>
+                    )}
+                  </>
+                )}
+              </div>
+            </div>
+
+            <button className={styles.entryButton} onClick={signOut}>
+              Sign out
+            </button>
+          </div>
+          <div className={styles.detailsSection}>
+            <div className={styles.row}>
+              <h2>Stytch Objects</h2>
+            </div>
+            <h3>Session</h3>
+            <pre className={styles.code}>{JSON.stringify(session, null, 2).replace(' ', '')}</pre>
+            <h3>User</h3>
+            <pre className={styles.code}>{JSON.stringify(user, null, 2).replace(' ', '')}</pre>
+          </div>
+        </div>
+      )}
+    </div>
+  );
+};
+
+export const getServerSideProps: GetServerSideProps = async (context) => {
+  // Get session from cookie
+  const cookies = new Cookies(context.req, context.res);
+  const storedSession = cookies.get('api_webauthn_session');
+  // If session does not exist display an error
+  if (!storedSession) {
+    return { props: { error: 'No user session found.' } };
+  }
+
+  try {
+    const stytchClient = loadStytch();
+    // Validate Stytch session
+    const { session } = await stytchClient.sessions.authenticate({ session_token: storedSession });
+    // Get the Stytch user object to display on page
+    const user = await stytchClient.users.get(session.user_id);
+
+    // Determine from the user object if this user has registered a webauthn device at this domain
+    const hasRegisteredWebAuthnDevice =
+      user.webauthn_registrations.length > 0 &&
+      user.webauthn_registrations.find((i) => i.domain === getStrippedDomain()) !== undefined;
+
+    // Determine if user has access to the super secret area data
+    let superSecretData = null;
+    if (
+      session.authentication_factors.length === 2 &&
+      session.authentication_factors.find((i) => i.delivery_method === 'email') &&
+      session.authentication_factors.find((i) => i.delivery_method === 'webauthn_registration')
+    ) {
+      superSecretData =
+        'Welcome to the super secret data area. If you inspect your Stytch session on the right you will see you have two authentication factors: email and webauthn_registration.';
```

Comment:
I'd include a third sentence here like, "You're only able to view the Super secret area because your session has both of these authentication factors."